src/libsyntax/parse/parser.rs

```diff
@@ -5218,8 +5218,25 @@ impl<'a> Parser<'a> {
             |p| {
                 let attrs = p.parse_outer_attributes()?;
                 let lo = p.span.lo;
-                let vis = p.parse_visibility(false)?;
-                let ty = p.parse_ty_sum()?;
+                let mut vis = p.parse_visibility(false)?;
+                let ty_is_interpolated =
+                    p.token.is_interpolated() || p.look_ahead(1, |t| t.is_interpolated());
```

Comment:
Do we have any tests covering this interpolated case? I expected to see one where we substitute in a `$t:ty`.

It is a bit of a shame that we have no `$p:path` fragment specificier, though I guess https://github.com/rust-lang/rfcs/pull/1575 would address the use case.